yarn-project/simulator/src/avm/avm_simulator.ts

```diff
@@ -65,8 +59,8 @@ export class AvmSimulator {
   public static create(
     stateManager: AvmPersistableStateManager,
     address: AztecAddress,
+    fnName: string,
```

Comment:
It doesn't seem right to initialize the simulator with the fnName. The name will _never_ be available to the sequencer, which is the most important user of the simulator. It will only be available locally via the PXE. Also, I suppose this is used mostly for debugging so it feels too much to ask for it.

I would recompute it internally using getDebugPublicFunctionName if needed.

Reply:
I changed this but since getDebudPublicFunctionName is asynchronous, we cannot perform it in a constructor.